ibis/pandas/core.py

```diff
@@ -243,19 +260,27 @@ def execute_until_in_scope(
     computable_args = [arg for arg in op.inputs if is_computable_input(arg)]
 
     # recursively compute each node's arguments until we've changed type
-    scopes = [
-        execute_until_in_scope(
-            arg,
-            new_scope,
-            aggcontext=aggcontext,
-            post_execute_=post_execute_,
-            clients=clients,
-            **kwargs,
+    if len(computed_localcontexts) == len(computable_args):
```

Comment:
I think you can use reorganize the code a little to make it more readable:

```
assert len(compute_localcontexts) == len(computable_args)
scopes = ...

```
or

```
if len(computed_localcontexts) == len(computable_args):
    raise ...

scopes = ...

```

Reply:
Sure, fixed